tests/darcy_velocity.prm

```diff
@@ -0,0 +1,148 @@
+#########################################################
+# This is a test for the new advection method that advects a
+# compositional field with the fluid velocity prescribed by
+# Darcy's Law.
+
+# The initial porosity is uniform everywhere, and in the
+# center of the model is a blob of 'fluid' and 'solid'.
+# The solid has a prescribed x velocity, and so the fluid
+# blob should be advected horizontally with the solid and
+# vertically as it rises up due to its buoyancy.
```

Comment:
I think some lines of documentation are also needed regarding the material model being used (melting_rate), which is defined in the test file `rising_melt_blob.cc`. Two-phase melt transport is turned off, but we are using the Melt interface to access and/or fill all the melt related terms. This will probably be confusing to anyone who is not familiar with the melt material model or how those work. 

After this PR is merged it would be good to add a benchmark that composites on top of the visco plastic material, and adds on all the melt-related terms.